File: srcJava/com/rti/perftest/harness/PerftestThreadPriorities.java
/*
 * (c) 2005-2017 Copyright, Real-Time Innovations, Inc. All rights reserved.
 * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
 */


package com.rti.perftest.harness;

import java.util.HashMap;
import java.util.Map;

public class PerftestThreadPriorities {

    public int main;
    public int receive;
    public int dbAndEvent;
    public boolean isSet;

    private static HashMap<Character, Integer> defaultPriorities = new HashMap<Character, Integer>();

    public PerftestThreadPriorities() {
        main = 0;
        receive = 0;
        dbAndEvent = 0;
        isSet = false;

        defaultPriorities.put('h', 99);
        defaultPriorities.put('n', 50);
        defaultPriorities.put('l', 1);

    }

}

File: srcCpp/perftestThreadPriorities.cxx
/*
 * (c) 2005-2018  Copyright, Real-Time Innovations, Inc. All rights reserved.
 * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
 */


#include "perftestThreadPriorities.h"

PerftestThreadPriorities::PerftestThreadPriorities()
        : main(0), receive(0), dbAndEvent(0), isSet(false)
{
    #ifdef RTI_WIN32
        defaultPriorities['h'] = THREAD_PRIORITY_TIME_CRITICAL;
        defaultPriorities['n'] = THREAD_PRIORITY_NORMAL;
        defaultPriorities['l'] = THREAD_PRIORITY_IDLE;
    #elif RTI_UNIX
        defaultPriorities['h'] = sched_get_priority_max(SCHED_FIFO);
        defaultPriorities['n'] = (sched_get_priority_max(SCHED_FIFO)
                + sched_get_priority_min(SCHED_FIFO)) / 2;
        defaultPriorities['l'] = sched_get_priority_min(SCHED_FIFO);
    #endif
}

bool PerftestThreadPriorities::set_priorities(char x, char y, char z)
{
    if (defaultPriorities.count(x) && defaultPriorities.count(y)
        && defaultPriorities.count(z)) {
        main = defaultPriorities[x];
        receive = defaultPriorities[y];
        dbAndEvent = defaultPriorities[z];
        return true;
    }
    return false;
}